bindings/py/cpp_src/bindings/engine/py_utils.hpp

```diff
@@ -41,6 +41,16 @@ namespace nupic_ext {
     template<typename T> T* get_end(py::array_t<T>& a) { return ((T*)a.request().ptr) + a.size(); }
 
     template<typename T> T* get_it(py::array& a) { return (T*)a.request().ptr; }
+
+    // Check that the precision in bytes matches the data size of the array
+    template<typename T> T* get_it( int precision, py::array& a)
+	{
+	if (precision != a.request().itemsize)
+		{throw std::invalid_argument("Invalid numpy array precision used.");}
+
+	return (T*)a.request().ptr;
```

Comment:
nit, if we are changing this, please prefer to use `return static_cast<T*>(a.request().ptr)`